buildSrc/src/main/kotlin/publish-conventions.gradle.kts

```diff
@@ -0,0 +1,42 @@
+plugins {
+    `maven-publish`
+    signing
+}
+
+// empty xxx-javadoc.jar
+val javadocJar = tasks.register<Jar>("javadocJar") {
```

Comment:
```suggestion
val javadocJar by tasks.registering(Jar::class) {
```

Reply:
I meet such error:
```
> Cannot convert the provided notation to an object of type MavenArtifact: org.gradle.kotlin.dsl.RegisteringDomainObjectDelegateProviderWithTypeAndAction@9a7d1ac.
  The following types/formats are supported:
    - Instances of MavenArtifact.
    - Instances of AbstractArchiveTask, for example jar.
    - Instances of PublishArtifact
    - Instances of Provider
    - Maps containing a 'source' entry, for example [source: '/path/to/file', extension: 'zip'].
    - Anything that can be converted to a file, as per Project.file()
```
in 
```
        if (name != "kotlinMultiplatform") { // The root module gets the JVM's javadoc JAR
            artifact(javadocJar)
        }
```
with such approach

Reply:
rewrote